### ISSUE TITLE
Compound value in context provider

### ISSUE THREAD

**Issue by XavierVal** [telefonicaid/fiware-orion#1440]
Sending a updateContext to CB to a field from a Context Provider, we have found that the value (that is a compound attribute) resent is an empty string

time=2015-10-28T15:09:08.108UTC | lvl=DEBUG | trans=N/A | function=ContextAttribute | comp=Orion | msg=ContextAttribute.cpp[100]: Creating a ContextAttribute: compoundValueP at (nil) for attribute 'op_result' at 0x7f28c000d8a0

**Update to CB: 
```
{
    "contextElements": [
        {
            "id": "xxx",
            "type": "aaaa",
            "isPattern": false,
            "attributes": [
                {
                    "name": "op_result",
                    "type": "string",
                    "value": {
                        "code": "200",
                        "extra": "rt-20%3Brrgb-00FF00%3B"
                    }
                }
            ]
        }
    ],
    "updateAction": "UPDATE"
}
```

**Update generated to the ContextProvier:
 ```
"contextResponses" : [
    {
      "contextElement" : {
        "type" : "aaaa",
        "isPattern" : "false",
        "id" : "xxx",
        "attributes" : [
          {
            "name" : "op_result",
            "type" : "string",
            "value" : ""  
          }
        ]
      },
      "statusCode" : {
        "code" : "200",
        "reasonPhrase" : "OK"
      }
    }
  ]
}
```

**Comment by kzangeli** [comment]
I understand this is a **POST /v1/updateContext**, right?

**Comment by fgalan** [comment]
No other payload in the whole API uses `updateAction`, so I guess so ;)

**Comment by XavierVal** [comment]
it is! POST /v1/updateContext

**Comment by fgalan** [comment]
I think the problem is related with this call in updateForward() function:

```
payload         = upcrP->render(ciP, UpdateContext, "");
```

(upcr stands for updateContextRequest, i.e. the payload generated to fill the HTTP request to forward).

So:

1. Maybe the problem is in the render() chain? I have tried to dig into in in order to find some point in which `compoundValueP == NULL` is not correctly checked, but I got lost :)
2. Maybe the problem is that the request that is used as updateForward() is not correctly filled? Unlikely, as I understand that this request comes directly from the one sent by the client at Orion API, but who knows...
